server/job/plexsync/index.ts

```diff
@@ -182,6 +231,68 @@ class JobPlexSync {
         if (matchedtmdb?.[1]) {
           tvShow = await this.tmdb.getTvShow({ tvId: Number(matchedtmdb[1]) });
         }
+      } else if (metadata.guid.match(hamaTvdbRegex)) {
+        const matched = metadata.guid.match(hamaTvdbRegex);
+        const tvdbId = matched?.[1];
+
+        if (tvdbId) {
+          tvShow = await this.tmdb.getShowByTvdbId({ tvdbId: Number(tvdbId) });
+          if (animeList.isLoaded()) {
+            await this.processHamaSpecials(plexitem, metadata, Number(tvdbId));
+          } else {
+            this.log(
+              `Hama id ${plexitem.guid} detected, but library agent is not set to Hama`,
+              'info'
+            );
+          }
+        }
+      } else if (metadata.guid.match(hamaAnidbRegex)) {
+        const matched = metadata.guid.match(hamaAnidbRegex);
+
+        if (!animeList.isLoaded()) {
+          this.log(
+            `Hama id ${plexitem.guid} detected, but library agent is not set to Hama`,
+            'info'
+          );
+        } else if (matched?.[1]) {
+          const anidbId = Number(matched[1]);
+          const result = await animeList.getFromAnidbId(anidbId);
+          if (result) {
+            // first try to lookup tvShow by tvdbid
+            if (result.tvdbId) {
+              const extResponse = await this.tmdb.getByExternalId({
+                externalId: result.tvdbId,
+                type: 'tvdb',
+              });
+              if (extResponse.tv_results[0]) {
+                tvShow = await this.tmdb.getTvShow({
+                  tvId: extResponse.tv_results[0].id,
+                });
+              }
+              await this.processHamaSpecials(plexitem, metadata, result.tvdbId);
+            }
+
+            // if getting tvShow above failed, try with tmdb/imdb for movie
+            if (tvShow == null) {
+              if (result.tmdbId) {
+                return await this.processMovieWithId(
+                  plexitem,
+                  undefined,
+                  result.tmdbId
+                );
```

Comment:
Maybe we should try to lookup by tmdb first if we have it? Then we can skip checking tvdb altogether.

Reply:
Majority of movies in mapping won't have tvdbid set, so previous lookup won't execute.
Only few movies have that - typically they are for anime that has tv show also (like Naruto).
I did tvdbid lookup first - then the problem is t hat your api calls to tmdb do not have code for TV lookup in tmdb. So tmdb lookup by tmdb id they would fail for tv shows - as mapping has tmdbid for tv show too. So instead of failing for majority tv shows lookups first by tmdbid in tmdb by and then succeeding by tvdbid in tmdb, I first check for tv show by tvdbid in tmdb - and only then for minority of movies first fail for tvdbid lookup in tmdb, and then succed by tmdbid lookup.

Reply:
Got it 👍